src/consumer/frame.js

```diff
@@ -10,6 +10,16 @@ import URI from '../lib/uri';
  */
 class Frame extends EventEmitter {
 
+  constructor(props) {
+    super(props);
+
+    // Binds 'this' for methods called internally
+    this.handleProviderMessage = this.handleProviderMessage.bind(this);
```

Comment:
Moving all of bindings out of init method. Following current ES6 coding pattern and putting them in constructor.